### DISCUSSION TITLE
Laravel Reverb Connection Established on Laravel end and not Client

### DISCUSSION THREAD

**Discussion by MHasanKN** [laravel/framework#51331] [General]
I did check out the other issues which were similar but they don't exactly quite fit my scenario. 

I am on a shared hosting trying to run Laravel Reverb in production. Locally it works fine.
Here is my .env:
`REVERB_APP_ID=504618
REVERB_APP_KEY=nelnxsrzsy6ijdqzkixe
REVERB_APP_SECRET=nic8gwfsl7cydwhgcu2n
REVERB_HOST="pogotrades.com"
REVERB_PORT=443
REVERB_SCHEME=https

VITE_REVERB_APP_KEY="${REVERB_APP_KEY}"
VITE_REVERB_HOST="${REVERB_HOST}"
VITE_REVERB_PORT="${REVERB_PORT}"
VITE_REVERB_SCHEME="${REVERB_SCHEME}"`

My .htaccess:
![image](https://github.com/laravel/framework/assets/106889523/583d0929-2d0c-4433-b8c3-af8e239acfa0)

You can see that I have commented out the 0.0.0.0 ip because I was testing both and both 127.0.01 and 0.0.0.0 result in same issue. Where I am stuck is that I ran this command using screen so that it never stops and it recieves a connection request, it says connection established but in front-end for client no response is received. After timeout the connection is closed by client and retried automatically and then I am stuck in a loop.

Below is laravel reverb:
![image](https://github.com/laravel/framework/assets/106889523/7ca4206d-570c-4f0e-ab0f-171d0a99960f)

Here is network request response after timeout:
![image](https://github.com/laravel/framework/assets/106889523/11c18551-41d1-4d9b-9b18-6b98479a4a2e)

Here is my Echo.js file:
```
import Echo from 'laravel-echo';

import Pusher from 'pusher-js';
window.Pusher = Pusher;

window.Echo = new Echo({
    broadcaster: 'reverb',
    key: import.meta.env.VITE_REVERB_APP_KEY,
    wsHost: import.meta.env.VITE_REVERB_HOST,
    wsPort: import.meta.env.VITE_REVERB_PORT ?? 80,
    wssPort: import.meta.env.VITE_REVERB_PORT ?? 443,
    forceTLS: (import.meta.env.VITE_REVERB_SCHEME ?? 'https') === 'https',
    enabledTransports: ['ws', 'wss'],
});
```

Here is my reverb.php:
```
<?php

return [
      'default' => env('REVERB_SERVER', 'reverb'),
'servers' => [

        'reverb' => [
            'host' => env('REVERB_SERVER_HOST', '0.0.0.0'),
            'port' => env('REVERB_SERVER_PORT', 8080),
            'hostname' => env('REVERB_HOST'),
            'options' => [
                'tls' => [],
            ],
            'max_request_size' => env('REVERB_MAX_REQUEST_SIZE', 10_000),
            'scaling' => [
                'enabled' => env('REVERB_SCALING_ENABLED', false),
                'channel' => env('REVERB_SCALING_CHANNEL', 'reverb'),
                'server' => [
                    'host' => env('REDIS_HOST', '127.0.0.1'),
                    'port' => env('REDIS_PORT', '6379'),
                    'password' => env('REDIS_PASSWORD'),
                    'database' => env('REDIS_DB', '0'),
                ],
            ],
            'pulse_ingest_interval' => env('REVERB_PULSE_INGEST_INTERVAL', 15),
            'telescope_ingest_interval' => env('REVERB_TELESCOPE_INGEST_INTERVAL', 15),
        ],

    ],
    'apps' => [

        'provider' => 'config',

        'apps' => [
            [
                'key' => env('REVERB_APP_KEY'),
                'secret' => env('REVERB_APP_SECRET'),
                'app_id' => env('REVERB_APP_ID'),
                'options' => [
                    'host' => env('REVERB_HOST'),
                    'port' => env('REVERB_PORT', 443),
                    'scheme' => env('REVERB_SCHEME', 'https'),
                    'useTLS' => env('REVERB_SCHEME', 'https') === 'https',
                ],
                'allowed_origins' => ['*'],
                'ping_interval' => env('REVERB_APP_PING_INTERVAL', 60),
                'max_message_size' => env('REVERB_APP_MAX_MESSAGE_SIZE', 10_000),
            ],
        ],

    ],

];

```
Any solution or idea what I am doing wrong or what's messing this stuff up?

**Comment by wilianmaique** [comment]
check this: https://stackoverflow.com/questions/78383337/solution-laravel-reverb-does-not-work-in-production-with-apache

**Reply by MHasanKN** [reply]
Thank you for the share but I did mention I am on shared hosting and can not edit files like ws-vhost.conf or ws-https-vhost.conf. The only thing that I can do is change the .htaccess but that didn't work for me. Changing config files requires sudo access which on shared hosting is not allowed.

**Reply by dp4kk** [reply]
I'm having similar issue. Did you finally find any solution to your problem?

**Reply by MHasanKN** [reply]
@dp4kk if you are on a shared hosting plan and they do not let you change the config files for ngnix or apache then there is no solution except going for a VPS and not a shared plan. I shifted all of this stuff to AWS and it works perfectly.
